### PR TITLE
Checkout merge commit for pull_request_target conformance suite action

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
#### Reason for change
Our conformance suite action was running on the target branch, without changes from master merged.

#### Description of change
`pull_request` actions checkout a merged version of the repository by default. Our conformance suite action uses `pull_request_target` (for security reasons) which does not have this default behavior. [This discussion](https://github.com/actions/checkout/issues/518#issuecomment-890401887) provides a solution.

#### Steps to Test
CI

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

File: .github/workflows/conformance-suites.yml
Comment (aaroncameron-wk) [reply .github/workflows/conformance-suites.yml:25]: Info [here](https://docs.github.com/en/free-pro-team@latest/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request) leads me to believe `merge_commit_sha` may be null until a test merge commit is successfully generated, which I assume would cause this checkout action to fail. 

In practice, there's enough of a delay between us pushing an update to a PR and approving the runs that we probably won't encounter incomplete mergeability checks, but I'll add a step to fail if `mergeable` is not `true`. If it fails, I believe we can just rerun that action after a short period once the mergeability checks complete and it should continue as normal.

**Review by derekgengenbacher-wf — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

**Pull request merged**
